packages/core/src/service/services/product.service.ts

```diff
@@ -195,7 +195,10 @@ export class ProductService {
             .createQueryBuilder('_product_translation')
             .select('_product_translation.baseId')
             .andWhere('_product_translation.slug = :slug', { slug });
-
+        // Add channel awareness
+        qb.innerJoin('product.channels', 'channel', 'channel.id = :channelId', {
```

Comment:
I believe this is not chaning much, since we just search of a product id for a slug first and then proceed to call 
`findOne` that interally calls `findOneInChannel` 🤔